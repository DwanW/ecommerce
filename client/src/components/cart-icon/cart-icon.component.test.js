import React from 'react';
import { shallow } from 'enzyme';

import { CartIcon } from './cart-icon.component';

describe('CartIcon component test', ()=> {
    let wrapper;
    let mockToggleCartHidden;

    beforeEach(()=> {
        mockToggleCartHidden = jest.fn();

        const mockProps = {
            itemCount: 0,
            toggleCartHidden: mockToggleCartHidden
        }

        wrapper = shallow(<CartIcon {...mockProps}/>)
    });

    it('should render CartIcon component', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('should call toggleCarthidden when clicked', ()=>{
        wrapper.find('CartContainer').simulate('click');
        expect(mockToggleCartHidden).toHaveBeenCalled();
    });

    it('should render count as text', ()=> {
        const count = Number(wrapper.find('ItemCountContainer').text());
        expect(count).toBe(0);
    });
});