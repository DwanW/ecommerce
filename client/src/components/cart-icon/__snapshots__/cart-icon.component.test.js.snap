// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component test should render CartIcon component 1`] = `ShallowWrapper {}`;
